File: Servlets/Interface/db2/src/java/dbase2/musicdb/insert_customer.java
package dbase2.musicdb;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletResponse;

import java.io.IOException;
import java.io.PrintWriter;

import java.sql.Connection;
import java.sql.SQLException;
import java.sql.Statement;
import java.sql.DriverManager;
import java.sql.ResultSet;

public class insert_customer extends HttpServlet{
  
  public void doGet(HttpServletRequest request, HttpServletResponse response)
    throws ServletException, IOException{
    response.setContentType("text/html");
    PrintWriter writer = response.getWriter();
    writer.write("<html>\n");
    writer.write("<head><title>Insert Customer</title></head>\n");
    writer.write("<body>\n");
    writer.write("<center>\n");
    
    String name = request.getParameter("name");
    if((name == null) || (name.length() == 0)){
      printMsg("Can not insert a customer with no name!", writer, request);
      return;
    }
    String city = request.getParameter("city");
    if((city == null) || (city.length() == 0)){
      printMsg("Can not insert a customer with no city!", writer, request);
      return;
    }
    
    writer.write("<hr style=\"margin-left:5em; margin-right:5em\">\n");
    writer.write("You issued the following INSERT query:<br>\n");
    writer.write("<ul>\n");
    writer.write("<li>Customer name: <em>"+name+"</em>\n");
    writer.write("<li>Customer city: <em>"+city+"</em>\n");
    writer.write("</ul>\n");
    writer.write("<hr style=\"margin-left:5em; margin-right:5em\">\n");
    writer.write("Database server response:<br>\n");
    
    try{
      Class.forName("com.mysql.jdbc.Driver");
    }catch(ClassNotFoundException exc){
      exc.printStackTrace();
      printMsg("No JDBC driver found!", writer, request);
      return;
    }
    try{
      Connection connection = DriverManager.getConnection("jdbc:mysql://localhost/db2", "student", "student");
      Statement statement = connection.createStatement(); //referential integrity
      ResultSet result = null;
      
      /*check if city already existing*/
       result = statement.executeQuery("SELECT city FROM city WHERE city = '"+city+"'");

       /*if city not existing insert in table city*/
       if (result == null)
       {
         /* Insert name into table city */
          String insert_sql_stmt = "INSERT INTO city VALUES('null', '"+city+"')";
          statement = connection.createStatement();
          int row = statement.executeUpdate(insert_sql_stmt);
       }
       
       /* Get cityid from table city */
       result = statement.executeQuery("SELECT cityid FROM city WHERE city = '"+city+"'");
       if (result == null)
         printMsg("database error!", writer, request);
       else
       {
          int cityid = new Integer(result.getInt("cityid"));
          
          /* Insert name, cityid into table customer */
          String insert_sql_stmt = "INSERT INTO customer VALUES('null', '"+cityid+"', '"+name+"')";
          statement = connection.createStatement();
          int row = statement.executeUpdate(insert_sql_stmt);
          writer.write("<strong>Customer inserted!</strong><br>\n");
        }
      printMsg("", writer, request);
      }catch(SQLException exc){
        exc.printStackTrace();
        printMsg("database error!", writer, request);
      }
}

  private void printMsg(String msg, PrintWriter writer, HttpServletRequest request){
    writer.write("<h1>" + msg + "</h1>\n");
    writer.write("</center>\n");
    writer.write("<hr style=\"margin-left:5em; margin-right:5em\">\n");
    writer.write("<center>\n");
    writer.write("<a href = \"insert_customer.html\">Back</a>\n");
    writer.write("</center>\n");
    writer.write("</body>\n");
    writer.write("</html>\n");
  }
}

File: PHP/Interface/insert_customer.php
<?
/* this script will handle the variables passed from the insert_customer.html file */

/* declare some relevant variables */

PRINT "<html>";
PRINT "<head><title>Insert Customer</title></head>";
PRINT "<body>";

include 'connection.php';

$name = $_POST["name"];
$city = $_POST["city"];

/* MySQL table created to store the data */
$table0 = "customer";
$table1 = "city";

$today = date("Y-m-d");
PRINT "<center>";
PRINT "Today is: $today.<br>";
PRINT "<hr style=\"margin-left:5em; margin-right:5em\">";
PRINT "You issued the following INSERT query:<br>";
PRINT "<ul>";
PRINT "<li>Customer name: <em>$name</em>";
PRINT "<li>Customer city: <em>$city</em>";
PRINT "</ul>";
PRINT "<hr style=\"margin-left:5em; margin-right:5em\">";
PRINT "Database server response:<br>"; 

/*check if city already existing*/
$query = "SELECT city FROM $table1 WHERE city = '$city'";
$result = MYSQL_QUERY($query);

/* How many rows in the result? */
if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

/*if city not existing insert in table city*/
if($number == 0)
{
  /* Insert name into table city */
  $query = "INSERT INTO $table1 VALUES('null', '$city')";
  $result = MYSQL_QUERY($query);

  if($result != 0)
  {
    $affected_rows =  mysql_affected_rows();
    PRINT "$table1: ";
    PRINT "<strong>Query OK. Rows affected $affected_rows</strong><br>";
  }
  else
  {
     PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
  }
} 

/* Get cityid from table city */
$query = "SELECT cityid FROM $table1 WHERE city = '$city'";
$result = MYSQL_QUERY($query);

if(!$result){
  $number = 0;
}else{
  $number = MYSQL_NUMROWS($result);
}

if ($number == 0)
{
  PRINT "<strong>Query FAILED. Probably you don't have the proper access rights!</strong><br>";
}
else
{
  $cityid = mysql_result($result, "0", "cityid");
  
  /* Insert name, cityid into table customer */
  $query = "INSERT INTO $table0 VALUES('null', '$cityid', '$name')";
  $result = MYSQL_QUERY($query);

  if($result != 0)
  {
     $affected_rows =  mysql_affected_rows();
     PRINT "$table0: ";
     PRINT "<strong>Query OK. Rows affected $affected_rows</strong><br>";
  }
}
PRINT "</center>";
PRINT "<hr style=\"margin-left:5em; margin-right:5em\">";
PRINT "<center>";
PRINT "<a href = \"insert_customer.html\">Back</a>";
PRINT "</center>";
PRINT "</body>";
PRINT "</html>";

/* Close the database connection */
MYSQL_CLOSE();

?>
